exporter/elasticsearchexporter/model.go

```diff
@@ -161,123 +157,31 @@ func (m *encodeModel) encodeLogECSMode(resource pcommon.Resource, record plog.Lo
 	return document
 }
 
-func (m *encodeModel) encodeMetrics(resource pcommon.Resource, metrics pmetric.MetricSlice, _ pcommon.InstrumentationScope) ([][]byte, error) {
-	var baseDoc objmodel.Document
-
-	baseDoc.AddAttributes("", resource.Attributes())
-
-	// Put all metrics that have the same attributes and timestamp in one document.
-	docs := map[uint32]*objmodel.Document{}
-	for i := 0; i < metrics.Len(); i++ {
-		metric := metrics.At(i)
-
-		var dps pmetric.NumberDataPointSlice
-
-		// Only Gauge and Sum metric types are supported at the moment.
-		switch metric.Type() {
-		case pmetric.MetricTypeGauge:
-			dps = metric.Gauge().DataPoints()
-		case pmetric.MetricTypeSum:
-			dps = metric.Sum().DataPoints()
-		}
-
-		for j := 0; j < dps.Len(); j++ {
-			dp := dps.At(j)
-
-			hash := metricHash(dp.Timestamp(), dp.Attributes())
-			doc, docExists := docs[hash]
-			if !docExists {
-				doc = baseDoc.Clone()
-				doc.AddTimestamp("@timestamp", dp.Timestamp())
-				doc.AddAttributes("", dp.Attributes())
-
-				docs[hash] = doc
-			}
-
-			switch dp.ValueType() {
-			case pmetric.NumberDataPointValueTypeDouble:
-				doc.AddAttribute(metric.Name(), pcommon.NewValueDouble(dp.DoubleValue()))
-			case pmetric.NumberDataPointValueTypeInt:
-				doc.AddAttribute(metric.Name(), pcommon.NewValueInt(dp.IntValue()))
-			}
-		}
-	}
-
-	res := make([][]byte, 0, len(docs))
-
-	for _, doc := range docs {
-		if m.dedup {
-			doc.Dedup()
-		} else if m.dedot {
-			doc.Sort()
-		}
-
-		var buf bytes.Buffer
-		err := doc.Serialize(&buf, m.dedot)
-		if err != nil {
-			return nil, err
-		}
+func (m *encodeModel) encodeMetricDataPoint(resource pcommon.Resource, _ pcommon.InstrumentationScope, metric pmetric.Metric, dataPoint pmetric.NumberDataPoint) ([]byte, error) {
```

Comment:
From reading the code, I didn't quite understand how grouping metrics with the same attributes to the same ES document works now that the hashing stuff is removed. Does each document contain a single metric/data point now or where is the grouping happening?

From a storage perspective, it's still best to group as many metrics into the same document as possible.

Reply:
> Does each document contain a single metric/data point now 

Yes, this is the state after removing hashing. To bring back grouping metrics in a document I would need to reimplement hashing in a different place in code (in exporter and not in  model). I can do this if needed, in this PR or in a separate one.

Reply:
Gotcha, let's do this in a follow-up PR then. Can you create an issue so that we can track this?

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33756.